### ISSUE TITLE
Provide interface stream out arrow RecordBatch

### ISSUE THREAD

**Issue by wangxiaoying** [sfu-db/connector-x#278]

**Comment by fnicastri** [comment]
it would be very helpful.
Any chances to see this in the future?

**Comment by wangxiaoying** [comment]
We have initialized the [arrow batch iterator](https://github.com/sfu-db/connector-x/blob/main/connectorx/src/arrow_batch_iter.rs) for rust and cpp library. Need more work in terms of testing and exposing to python library.

**Comment by chitralverma** [comment]
I think it will be an awesome addition to be able to get a [RecordBatchReader](https://arrow.apache.org/docs/python/generated/pyarrow.RecordBatchReader.html) directly from `read_sql` which only materializes the record batches (sends queries to DB) when the user requests `read_next_batch`. 

@wangxiaoying you think the testing is done to expose arrow record batch iterator to python side, is there anyway I can help with this?

**Comment by wangxiaoying** [comment]
> I think it will be an awesome addition to be able to get a [RecordBatchReader](https://arrow.apache.org/docs/python/generated/pyarrow.RecordBatchReader.html) directly from `read_sql` which only materializes the record batches (sends queries to DB) when the user requests `read_next_batch`.
> 
> [@wangxiaoying](https://github.com/wangxiaoying) you think the testing is done to expose arrow record batch iterator to python side, is there anyway I can help with this?

Yes, I think we can definitely enable the record batch reader. Please feel free to submit a PR!

**Comment by chitralverma** [comment]
actually, scratch the above, I managed to get this working exactly as expected. will raise a PR today for your review. :D
